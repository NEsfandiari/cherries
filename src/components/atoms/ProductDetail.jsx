import React, { Component } from 'react'
import styled from 'styled-components'

const Layout = styled.div`
  font-size: 0.8rem;
`

class ProductDetail extends Component {
  constructor(props) {
    super(props)
  }
  render() {
    const details = this.props.details.map(d => <p>{d}</p>)
    return (
      <Layout>
        <h4>Details</h4>
        {details}
      </Layout>
    )
  }
}

export default ProductDetail
