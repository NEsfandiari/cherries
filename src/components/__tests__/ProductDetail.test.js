import React from 'react'
import renderer from 'react-test-renderer'
import ProductDetail from '../atoms/ProductDetail'

describe('ProductDetail', () =>
  it('renders correctly', () => {
    const details = [
      'Materials: something something something',

      'Closure: something else',

      'Allergy Information: something the most',
    ]
    const tree = renderer.create(<ProductDetail details={details} />).toJSON()
    expect(tree).toMatchSnapshot()
  }))
