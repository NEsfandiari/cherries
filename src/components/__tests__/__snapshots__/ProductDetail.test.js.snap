// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetail renders correctly 1`] = `
<div
  className="sc-bdVaJa hRKdxn"
>
  <h4>
    Details
  </h4>
  <p>
    Materials: something something something
  </p>
  <p>
    Closure: something else
  </p>
  <p>
    Allergy Information: something the most
  </p>
</div>
`;
